faucet.go

```diff
@@ -456,7 +465,14 @@ func (l *lightningFaucet) faucetHome(w http.ResponseWriter, r *http.Request) {
 	// form to open a channel, so we'll pass that off to the openChannel
 	// handler.
 	case r.Method == http.MethodPost:
-		l.openChannel(homeTemplate, homeInfo, w, r)
+		action, _ := r.URL.Query()["action"]
+
+		// action == 0 is to open channel, action == 1 is to generate
+		if action[0] == "0" {
```

Comment:
Don't use a magic number. Use a string constant somewhere ("openChannel", "generateInvoice") and pass it onto the template.